crates/cxx-qt-gen/src/generator/rust/signals.rs

```diff
@@ -43,10 +43,13 @@ pub fn generate_rust_signals(
             })
             .collect::<Vec<TokenStream>>();
 
-        let self_type = if signal.mutable {
-            quote! { Pin<&mut #qobject_name> }
+        let (self_type, self_type_cxx) = if signal.mutable {
+            (
+                quote! { core::pin::Pin<&mut #qobject_name> },
```

Comment:
once #553 is ready we'll have a method we can just pass types through to auto add fully qualified syntax.